ProjectLighthouse/Database.cs

```diff
@@ -312,6 +317,40 @@ public async Task UnqueueLevel(int userId, Slot queuedSlot)
         await this.SaveChangesAsync();
     }
 
+    public async Task BlockUser(int userId, User blockedUser)
+    {
+        if (userId == blockedUser.UserId)
+        {
+            return;
+        }
+        
+        User? user = await this.Users.FirstOrDefaultAsync(u => u.UserId == userId);
+        
+        BlockedProfile blockedProfile = new()
+        {
+            User = user,
+            BlockedUser = blockedUser,
+        };
+
+        await this.BlockedProfiles.AddAsync(blockedProfile);
+
+        await this.SaveChangesAsync();
+    }
+
+    public void UnblockUser(int userId, User blockedUser)
+    {
+        if (userId == blockedUser.UserId) return;
+
+        this.BlockedProfiles.RemoveWhere(bp => bp.BlockedUser == blockedUser && bp.UserId == userId);
+    }
+
+    public async Task<bool> IsBlocked(int blockedUserId, int userId)
+    {
+        if (userId == blockedUserId) return false;
+
+        return await this.BlockedProfiles.Has(bp => bp.BlockedUserId == blockedUserId && bp.UserId == userId);
+    }
```

Comment:
```suggestion
    public async Task<bool> IsUserBlockedBy(int userId, int targetId)
    {
        if (userId == targetUser) return false;

        return await this.BlockedProfiles.Has(bp => bp.BlockedUserId == userId && bp.UserId == targetUser);
    }
```
The IsBlocked naming is kind of ambiguous as to what it is checking in relation to the parameters. I think a more descriptive name could help a little along with renaming the parameters to go along with the name.

Reply:
This looks wrong, shouldn't it be targetId on line 348

Reply:
Yeah targetUser should be targetId